Include the model name when filtering encrypted attributes.

jorgemanrubia opened the pull request:
For example, when encrypting `Person#name` it will add `person.name` as a filter parameter, instead of just `name`. This prevents unintended filtering of parameters with a matching name in other models.

Addresses the second concern in https://github.com/rails/rails/issues/44330.

CC @rafaelfranca @nvasilevski

jorgemanrubia committed:
Include the model name when filtering encrypted attributes.

For example, when encrypting `Person#name` it will add `person.name` as a filter
parameter, instead of just `name`. This prevents unintended filtering of parameters
with a matching name in other models.

Closes https://github.com/rails/rails/issues/44330

nvasilevski commented on activerecord/lib/active_record/encryption/configurable.rb:
```diff
@@ -52,7 +52,8 @@ def encrypted_attribute_was_declared(klass, name) # :nodoc:
         def install_auto_filtered_parameters_hook(application) # :nodoc:
           ActiveRecord::Encryption.on_encrypted_attribute_declared do |klass, encrypted_attribute_name|
-            application.config.filter_parameters << encrypted_attribute_name unless ActiveRecord::Encryption.config.excluded_from_filter_parameters.include?(encrypted_attribute_name)
+            filter_parameter = [("#{klass.model_name.element}." if klass.name), encrypted_attribute_name.to_s].compact.join
```
It's a nitpick really but I feel like the "dot" is a little bit hidden so perhaps we could make it a little explicit, like:
```suggestion
            filter_parameter = [("#{klass.model_name.element}" if klass.name), encrypted_attribute_name.to_s].compact.join(".")
```

jorgemanrubia replied:
Thanks @nvasilevski I grabbed that :) 👍

nvasilevski reviewed (commented):
Thanks ❤️ 
The change looks good

jorgemanrubia commented:
@rafaelfranca yes, you are correct. It will work as long as the resource/param name matches the model/attribute name. I think it's a sensible default, even knowing it won't work in all cases. The previous default was probably too coarse grained. I rebased and fixed conflicts.

rafaelfranca commented:
👍🏽 . There is a failing test though.

nvasilevski commented:
just wanted to bump this PR. Sorry for the noise
The failure does seem unrelated but perhaps worth rebasing just to have a green CI
